### PR TITLE
TaskDialogCallback improvment

### PR DESCRIPTION
**Pull request by derChristoph**
Added RADIO_BUTTON_CLICKED, HYPERLINK_CLICKED and VERIFICATION_CLICKED to the Callback routine

### CODE REVIEW

**Comment by synopse** [comment]
Your commit add a dependency to ShellExecute which is not a good idea in some context...

File: SynTaskDialog.pas [hunk 1]
```diff
@@ -653,23 +653,36 @@ function CR(const aText: string): string;
...
       CanClose := True;
       ptd^.Dialog.OnButtonClicked(ptd,wParam,CanClose);
       if not CanClose then
         Result := S_FALSE;
     end;
+    TDN_HYPERLINK_CLICKED:
+    begin
+      sHyperLink := pChar(lParam);
+      if sHyperLink <> '' then
+        ShellExecute(Application.Handle, 'open', PChar(sHyperLink), nil, nil, SW_SHOW);
```
Comment (leus) [inline SynTaskDialog.pas]: Better add some form of callback for this

File: SynTaskDialog.pas [hunk 2]
```diff
@@ -1227,4 +1240,4 @@ finalization
   DefaultFont.Free;
   BitmapArrow.Free;
   BitmapOK.Free;
-end.
+end.
```
Comment (leus) [inline SynTaskDialog.pas]: ouch - no newline at the end of the file

**Comment by synopse** [comment]
Outdated with no feedback.

Please reopen if needed.

**Pull request closed without merging**